src/ldp/http/metadata/LinkParser.ts

```diff
@@ -1,17 +1,27 @@
+import type { NamedNode } from '@rdfjs/types';
 import { DataFactory } from 'n3';
 import { getLoggerFor } from '../../../logging/LogUtil';
 import type { HttpRequest } from '../../../server/HttpRequest';
 import { parseParameters, splitAndClean, transformQuotedStrings } from '../../../util/HeaderUtil';
-import { RDF } from '../../../util/Vocabularies';
 import type { RepresentationMetadata } from '../../representation/RepresentationMetadata';
 import { MetadataParser } from './MetadataParser';
 
 /**
- * Parses Link headers with "rel=type" parameters and adds them as RDF.type metadata.
+ * Parses Link headers and adds them as metadata with the given predicate.
  */
-export class LinkTypeParser extends MetadataParser {
+export class LinkParser extends MetadataParser {
   protected readonly logger = getLoggerFor(this);
 
+  private readonly linkMap: Record<string, NamedNode>;
+
+  public constructor(linkMap: Record<string, string>) {
+    super();
+    this.linkMap = {};
+    Object.keys(linkMap).forEach((key: string): void => {
+      this.linkMap[key] = DataFactory.namedNode(linkMap[key]);
+    });
```

Comment:
```suggestion
    this.linkMap = Object.fromEntries(Object.entries(linkMap).map(([ header, uri ]) => [ header, namedNode(uri) ]))
```
or
```suggestion
    this.linkMap = {};
    for (const [ header, uri ] of Object.entries(linkMap) {
      this.linkMap[header] = namedNode(uri);
    }
```
Either way is fine, we just prefer not to use `Array.foreach` calls.

You can have `import namedNode = DataFactory.namedNode;` in your imports if you don't want to copy DataFactory here.